### DISCUSSION TITLE
HEIC Converter

### DISCUSSION THREAD

**Discussion by gm90** [johnlindquist/kit#174] [Share]
Loving Kit! ❤️  Awesome work @johnlindquist 🔥 

The simplicity of Airdrop between Apple devices is great until your image turns up in a file format that most file uploads still don't support. Used to spend a few mins opening up preview and re-exporting. So wrote this little script to do the heavy lifting.

```javascript
// Menu: HEIC Converter
// Description: Convert the currently selected HEIC file to either JPEG or PNG
// Author: Gary McCann

const { getSelectedFile } = await kit("file");
const { promises: fs } = await npm("fs");
const path = await npm("path");
const convert = await npm("heic-convert");

const imagePath = await getSelectedFile();
const { dir, name: imageName, ext } = path.parse(imagePath);

const format = await arg(
  "Which format would you like to convert to:",
  async () => {
    let options = ["JPEG", "PNG"];

    return options.map((format) => {
      return {
        name: format,
        value: format,
      };
    });
  }
);

const buffer = await fs.readFile(imagePath);
const outputBuffer = await convert({
  buffer,
  format,
  quality: 1,
});

const updatedFilePath = `${dir}/${imageName}.${format}`;

await fs.writeFile(updatedFilePath, outputBuffer);
```

Few bits to add:

1. Validation to make sure a HEIC file is selected.
2. Would like to maybe show some sort of progress bar, whilst its doing its business.

**Comment by johnlindquist** [comment]
Love it! Glad it helped remove some friction from your day ❤️

1. I think [file-type](https://www.npmjs.com/package/file-type) could help with detecting heic
2. Here's a cheap "busy" indicator 😉
[Install testing-progress-in-hint](https://scriptkit.app/api/new?name=testing-progress-in-hint&url=https://gist.githubusercontent.com/johnlindquist/1f293b9c446ce9fc10e8e2f0fe9aabaa/raw/bc571d477f2828015ee7c2d715f7675911750d5e/testing-progress-in-hint.js)

```js
let dots = "."
let id = setInterval(() => {
  setHint(dots)
  dots += "."
  if (dots.length > 5) dots = "."
}, 100)

await wait(5000)
clearInterval(id)

await arg("Stop")
```
